File: outputs/discord_webhook.py
import requests
from outputs.output_interface import OutputInterface


class DiscordWebhook(OutputInterface):
    def __init__(self, webhook_url: str, username: str = "aya"):
        self.webhook_url = webhook_url
        self.username = username
        self.buff = ""

    def print(self, text: str):
        self.buff = self.buff + text

    def flush(self):
        payload = {'content': self.buff,
                   'username': self.username}
        files = [

        ]
        headers = {}
        response = requests.request("POST", self.webhook_url, headers=headers, data=payload, files=files)
        self.buff = ""

File: internal/output/discord_webhook.go
package output

import (
	"aya-money-go/internal/contracts"
	"aya-money-go/internal/utils"
)

type output struct {
	webhookUrl string
	buff       string
}

type dcExecutePayload struct {
	Content  string `json:"content"`
	Username string `json:"username"`
}

func (o *output) Print(text string) error {
	o.buff += text
	return nil
}

func (o *output) Flush() error {
	body := &dcExecutePayload{
		Content:  o.buff,
		Username: "aya",
	}
	err := utils.SendJsonRequest("POST", o.webhookUrl, body, nil)
	if err != nil {
		return err
	}

	o.buff = ""
	return nil
}

func NewOutputDiscordWebhook(webhookUrl string) contracts.Output {
	return &output{webhookUrl: webhookUrl, buff: ""}
}
